Develop

Sinuousity opened the pull request:
Same as last closed pull request... except this one works.

Sinuousity committed:
Added Light animation based on LIGHRecord flags. Changed light object creation to remove bounce intensity (it's not supported anyways). Temporarily enabled exterior cell lights.

Sinuousity committed:
Implemented a local settings method.

This should make it much easier for different developers who have different install directories and preferred settings.
Instanced settings files should not be pushed, but LocalSettingsObject.cs should be.

New asset can be created through Assets>Create>TES Unity>Local Settings
Settings file needs to be inside a Resources folder so that the TESUnity instance can find it easily.

Sinuousity committed:
Simple fix for when no settings files are found.

Sinuousity committed:
Removed interaction with objects that have no formal name.
Implemented local settings render path.

ColeDeanShepherd commented on Assets/Scripts/TES/MorrowindEngine.cs:
```diff
@@ -301,20 +302,23 @@ public void CastInteractRay()
 					if ( component != null )
 					{
-						switch ( component.gameObject.tag )
+						if ( !string.IsNullOrEmpty( component.objData.name ) )
```
Wow, somehow I never knew IsNullOrEmpty existed...

Sinuousity replied:
I was having problems with object interaction where objects had white space names. Very handy!

ColeDeanShepherd commented on Assets/Scripts/TES/MorrowindEngine.cs:
```diff
@@ -551,9 +556,10 @@ private GameObject InstantiateLight(LIGHRecord LIGH, bool indoors)
 			//lightComponent.shadows = LightShadows.Soft;
 
-			if(!indoors)
+			if(!indoors & false)//disabling exterior cell lights because there is no day/night cycle
```
I'm confused by this line: doesn't the condition always evaluate to false, resulting in lights never being disabled?

Sinuousity replied:
That's correct. I was using this temporarily but I've already added the option to the local settings so you can can toggle it from there instead.

ColeDeanShepherd commented:
Everything looks good aside from the "if(!indoors & false)" line.

Sinuousity committed:
Big change - adding kinematic rigidbodies to all static colliders.

I would like extra confirmation that this doesn't cause issues in other areas, but adding rigidbodies to all static colliders prevents the collision tree from being rebuilt. This improves cell instantiation speed and improves framerate while objects are being created.

Smaller changes
- added exterior cell lights toggle to Settings file
-separated Settings file options into categories - will most likely be expanded to have more categories and I will write a custom inspector to make it look nicer

Sinuousity commented:
Wondering if I should close this request (again) and put in a new one with the new fixes.

Sinuousity committed:
Added the remaining object types interaction support. Should be all of them but let me know if there are any left out, they're easy to add now.

Increased the fly-camera speed because now cells load fast enough to warrant it.

Sinuousity committed:
Fix for weird bug: certain containers not tagged

Sinuousity committed:
no message

Sinuousity committed:
Added new Settings options for:
-Light Animations
-Sun Shadows
-Light Shadows

Improved the way Settings options are handled by TESUnity and other classes.

Fixed a sneaky issue involving lights in which many lights would not correctly find their Light component. This caused affected lights to discard any light flags in their LIGHRecord. This is now fixed.
Flickering lights are not a bug. The LightAnim behaviors just need tweaking.

Sinuousity committed:
Added ambient light intensity and a toggle for the kinematic rigidbody fix to the Settings options.
Fixes.

Sinuousity committed:
Improved material shaders.

Consolidated functionality to one CGInclude file and corrected assignment of alpha tested material shaders.
